services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/Hakemusalue.kt

```diff
@@ -74,3 +110,34 @@ fun List<KaivuilmoitusAlue>?.combinedAddress(): PostalAddress? =
         ?.joinToString(", ")
         ?.let { StreetAddress(it) }
         ?.let { PostalAddress(it, "", "") }
+
+private fun checkChangesInTyoalueet(
+    path: String,
+    firstAreas: List<Tyoalue>,
+    secondAreas: List<Tyoalue>,
+): List<String> {
+    val workAreaPath = "$path.tyoalueet"
+    val changedElementsInFirst =
+        firstAreas.withIndex().flatMap { (i, tyoalue) ->
+            val otherTyoalue = secondAreas.getOrNull(i)
+            if (otherTyoalue == null) {
+                listOf("$workAreaPath[$i]")
+            } else {
+                if (tyoalue.checkChange(Tyoalue::geometry, otherTyoalue) != null) {
+                    listOf("$workAreaPath[$i]", "$workAreaPath[$i].geometry")
+                } else emptyList()
+            }
```

Comment:
`checkChange` is just an overly complicated equals if the returned property name isn't used. It could be replaced with a basic equals here.

Extra nesting can be removed by using `else if`.

```suggestion
            } else if (tyoalue.geometry != otherTyoalue.geometry) {
                listOf("$workAreaPath[$i]", "$workAreaPath[$i].geometry")
            } else emptyList()
```

Reply:
Good point, thank you. Changed as suggested.